### DISCUSSION TITLE
Essence versus Package - how to achieve this

### DISCUSSION THREAD

**Discussion by david3a** [vsf-tv/gccg-api#34] [General]
If we want to support a package of essence how do we do this, 

Initial MVP will target single essence.

**Comment by gkelly1** [comment]
I have been thinking a little more on this from and I think from a media exchange perspective we can accomplish packaging data from different sources by positioning the data within a single buffer and using the attached metadata to provide information on location of data within the buffer, size of data etc... I think this could be done without having to modify the API required for a single essence. 

Smaller number of larger transfers will generally be more performant than larger number of smaller transactions. I could see this capability being useful for intra service data sharing, e.g. where service/application needs to scale across pods/nodes etc... It may not turn out not to be an issue but scalability could be a concern as number of unique connections grows. I am doubtful that it will be but I think it would be beneficial if we considered providing this flexibility within the media exchange layer even if  it is not part of the MVP.

**Reply by david3a** [reply]
Could this be aided by creating a package or group that allows the API to understand that these are related and then manage memory allocations in the manner you describe? That way, we get the best of both worlds.
